arangod/Graph/ShortestPathOptions.cpp

```diff
@@ -212,18 +217,42 @@ auto ShortestPathOptions::estimateDepth() const noexcept -> uint64_t {
   return 7;
 }
 
+auto ShortestPathOptions::setDefaultWeight(double weight) -> void {
+  if (weight <= 0.) {
+    THROW_ARANGO_EXCEPTION_MESSAGE(
+        TRI_ERROR_GRAPH_INVALID_EDGE_WEIGHT,
+        "negative or zero value default weight not allowed");
+  }
+  _defaultWeight = weight;
+}
+
+auto ShortestPathOptions::setWeightAttribute(std::string attribute) -> void {
```

Comment:
I suggest we avoid unnecessary string copy

Reply:
Done in: eebdeb77e47ac1e4ad2e86e587288a3c0cc4da28
(also included in devel port)